src/main/java/io/cdap/plugin/gcp/dataplex/common/util/DataplexUtil.java

```diff
@@ -0,0 +1,203 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.gcp.dataplex.common.util;
+
+import com.google.cloud.bigquery.Field.Mode;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.etl.api.FailureCollector;
+import io.cdap.plugin.gcp.bigquery.util.BigQueryTypeSize;
+import io.cdap.plugin.gcp.bigquery.util.BigQueryUtil;
+import io.cdap.plugin.gcp.dataplex.common.model.Field;
+import io.cdap.plugin.gcp.dataplex.sink.DataplexBatchSink;
+import io.cdap.plugin.gcp.dataplex.source.DataplexBatchSource;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nullable;
+
+/**
+ * Common Util class for dataplex plugins such as {@link DataplexBatchSource} and {@link DataplexBatchSink}
+ */
+public final class DataplexUtil {
+
+  /**
+   * Converts Entity Schema into a CDAP Schema object.
+   *
+   * @param dataplexSchema Dataplex Schema to be converted.
+   * @param collector      Failure collector to collect failure messages for the client.
+   * @return CDAP schema object
+   */
+  public static Schema getTableSchema(io.cdap.plugin.gcp.dataplex.common.model.Schema dataplexSchema,
+                                      @Nullable FailureCollector collector) {
+    List<Field> fields = dataplexSchema.getFields();
+    if (dataplexSchema.getPartitionFields() != null && !dataplexSchema.getPartitionFields().isEmpty()) {
+      fields.addAll(dataplexSchema.getPartitionFields());
+    }
+    List<Schema.Field> schemafields = new ArrayList<>();
+
+    for (Field field : fields) {
+      Schema.Field schemaField = getSchemaField(field, null, null);
+      // if schema field is null, that means that there was a validation error. We will still continue in order to
+      // collect more errors
+      if (schemaField == null) {
+        continue;
+      }
+      schemafields.add(schemaField);
+    }
+    if (schemafields.isEmpty() && collector != null && !collector.getValidationFailures().isEmpty()) {
+      // throw if there was validation failure(s) added to the collector
+      collector.getOrThrowException();
+    }
+    if (schemafields.isEmpty()) {
+      return null;
+    }
+    return Schema.recordOf("output", schemafields);
+  }
+
+  /**
+   * Converts Entity schema field into a corresponding CDAP Schema.Field.
+   *
+   * @param field        BigQuery field to be converted.
+   * @param collector    Failure collector to collect failure messages for the client.
+   * @param recordPrefix String to prepend to recordNames to make them unique
+   * @return A CDAP schema field
+   */
+  @Nullable
+  public static Schema.Field getSchemaField(Field field,
```

Comment:
is getSchemaField() called from outside this class? if not, no need to make it public

Reply:
done